src/Raven.Client/Documents/Session/InMemoryDocumentSessionOperations.cs

```diff
@@ -1278,6 +1279,34 @@ public void RegisterMissingIncludes(BlittableJsonReaderArray results, BlittableJ
             }
         }
 
+        internal void RegisterCounters(BlittableJsonReaderArray counters)
+        {
+            if (counters == null)
+                return;
+
```

Comment:
You need to account for counters that you _tried_ to load and wasn't able to.
Also, don't you need to account for `GotAll` too?

Reply:
fixed